pkg/appgw/backendhttpsettings.go

```diff
@@ -167,22 +167,29 @@ func (c *appGwConfigBuilder) getBackendsAndSettingsMap(cbCtx *ConfigBuilderConte
 
 	// enforce single pair relationship between service port and backend port
 	for backendID, serviceBackendPairs := range serviceBackendPairsMap {
+
+		// in case there are multiple backend ports found, using the smallest port in http setting
+		var backendport Port
+		backendport = 65536
+		var uniquePair serviceBackendPortPair
+		for k := range serviceBackendPairs {
+			if k.BackendPort <= backendport {
+				uniquePair = k
+				backendport = k.BackendPort
+			}
+		}
+
 		if len(serviceBackendPairs) > 1 {
+
 			// more than one possible backend port exposed through ingress
 			e := controllererrors.NewErrorf(
 				controllererrors.ErrorMultipleServiceBackendPortBinding,
-				"service:port [%s:%s] has more than one service-backend port binding",
-				backendID.serviceKey(), backendID.Backend.ServicePort.String(),
+				"service:port [%s:%s] has more than one service-backend port binding which is not an ideal scenario, choosing the smallest service-backend port %d",
```

Comment:
can  we also print the other ports here ?

Reply:
You mean endpoint ports matched with the service endpoint name in case of multiple ports?